yaks/lib/yaks/reader/hal.rb

```diff
@@ -29,7 +29,7 @@ def convert_links(links)
             rel = rel.to_sym if rel =~ /\A\w+\z/
             Resource::Link.new(rel: rel, uri: l['href'], options: options)
           end
-        end.to_set
+        end
```

Comment:
Ok, when I saw the `to_set` call I first thought about uniqueness rather than ordering since two identical links does not make sense as you said.